### PR TITLE
Moves Goonchat chat debug logging to RUSTG

### PR DESCRIPTION
**Pull request by dearmochi**
## What Does This PR Do

- Moves Goonchat chat debug logging to RUSTG

## Why It's Good For The Game

- It used direct output which sucks.

## Changelog
N/A

### CODE REVIEW

**Comment by AffectedArc07** [comment]
For context of what this is nuking
image

Writing to this thing is super expensive. It needs to be split.

File: goon/code/datums/browserOutput.dm [hunk 1]
```diff
@@ -202,8 +203,7 @@ var/list/chatResources = list(
 	return "pong"
 
 /datum/chatOutput/proc/debug(error)
-	error = "\[[time2text(world.realtime, "YYYY-MM-DD hh:mm:ss")]\] Client : [owner.key ? owner.key : owner] triggered JS error: [error]"
-	chatDebug << error
+	rustg_log_write(CHAT_DEBUG_LOG, "Client : [owner.key || owner] triggered JS error: [error][GLOB.log_end]")
```
Comment (AffectedArc07) [inline goon/code/datums/browserOutput.dm:206]: Could you also split it by round and put it in `SetupLogs()` in `world/New()`

It cuts down on bigass files, and also means denghis can do parsing magic on it per round.

File: goon/code/datums/browserOutput.dm [hunk 2]
```diff
@@ -92,7 +91,10 @@ var/list/chatResources = list(
 			data = doneLoading(arglist(params))
 
 		if("debug")
-			data = debug(arglist(params))
+			if(!length(params))
+				return
+			var/error = params[1]
+			data = log_chat_debug("Client: [owner.key || owner] triggered JS error: [error][GLOB.log_end]")
```
Comment (AffectedArc07) [inline goon/code/datums/browserOutput.dm:97]: Any reason for the assign here?

Comment (dearmochi) [inline goon/code/datums/browserOutput.dm:97]:
```suggestion
			log_chat_debug("Client: [owner.key || owner] triggered JS error: [error][GLOB.log_end]")
```

You utter donkey

**Review by AffectedArc07 — APPROVED** [review]

**Pull request merged**